docs-core/src/main/java/com/sismics/docs/core/service/InboxService.java

```diff
@@ -235,6 +264,33 @@ private void importMessage(Message message) throws Exception {
             FileUtil.createFile(fileContent.getName(), null, fileContent.getFile(), fileContent.getSize(),
                     document.getLanguage(), "admin", document.getId());
         }
+
+        if (ConfigUtil.getConfigBooleanValue(ConfigType.INBOX_DELETE_IMPORTED)) {
+            message.setFlag(Flags.Flag.DELETED, true);
+        }
+    }
+
+    /**
+     * Fetches a HashMap with all tag names in lowercase as keys and their respective ids as values.
+     *
+     * @return HashMap with all tags or null if not enabled
+     */
+    private HashMap<String, String> getAllTags() {
+        if (!ConfigUtil.getConfigBooleanValue(ConfigType.INBOX_AUTOMATIC_TAGS)) {
+            return null;
+        }
+        TagDao tagDao = new TagDao();
+        List<TagDto> tags = tagDao.findByCriteria(new TagCriteria().setTargetIdList(Lists.newArrayList("admin")), new SortCriteria(1, true));
+
+        HashMap<String, String> tagsNameToId = new HashMap<>();
+        for (TagDto tagDto : tags) {
+            if (tagsNameToId.containsKey(tagDto.getName())) {
+                tagsNameToId.put(tagDto.getName(), null);
```

Comment:
Why put a null in the map if the tag was already added? (should not happen as tag names are unique by the way).

Reply:
Okay, if they are unique this can be removed